src/fides/api/models/privacy_experience.py

```diff
@@ -97,38 +255,131 @@ def create(
+
         experience_config: PrivacyExperienceConfig = super().create(
             db=db, data=data, check_name=check_name
         )
-        # create the history after the initial object creation succeeds, to avoid
-        # writing history if the creation fails and so that we can get the generated ID
-        data.pop(
-            "id", None
-        )  # Default Experience Configs have id specified but we don't want to use the same id for the historical record
-        history_data = {
-            **data,
-            "experience_config_id": experience_config.id,
-        }
-        PrivacyExperienceConfigHistory.create(db, data=history_data, check_name=False)
+
+        for translation_data in translations:
+            # Create an ExperienceTranslation
+            translation = ExperienceTranslation.create(
+                db,
+                data={**translation_data, "experience_config_id": experience_config.id},
+            )
+            # Version the ExperienceTranslation and the original PrivacyExperienceConfig together.
+            PrivacyExperienceConfigHistory.create(
+                db,
+                data={**data, **translation_data, "translation_id": translation.id},
+                check_name=False,
+            )
+
+        # Link regions to this PrivacyExperienceConfig via the PrivacyExperience table
+        upsert_privacy_experiences_after_config_update(db, experience_config, regions)
+        # Link Privacy Notices to this Privacy Experience config via the ExperienceNotices table
+        link_notices_to_experience_config(
+            db, notice_ids=privacy_notice_ids, experience_config=experience_config
+        )
+
         return experience_config
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperienceConfig:
         """
-        Overrides the base update method to automatically bump the version of the
-        PrivacyExperienceConfig record and also create a new PrivacyExperienceConfigHistory entry
+        Updates a PrivacyExperienceConfig and related resources which is a large exercise!
+
+        - Updates the PrivacyExperienceConfig details
+        - For each supplied translation, add, update, or delete translations so they match
+        the translations in the update request
+        - For each remaining translation, if the translation or the config has changed,
+        create a historical record for auditing purposes
+        - Link or unlink regions (via Privacy Experiences)
+        - Link or unlink Privacy Notices
         """
-        resource, updated = update_if_modified(self, db=db, data=data)
+        request_translations = data.pop("translations", [])
+        regions = data.pop("regions", [])
+        privacy_notice_ids = data.pop("privacy_notice_ids", [])
+
+        # Do a patch update of the existing privacy experience config if applicable
+        config_updated = update_if_modified(self, db=db, data=data)
```

Comment:
yep, I can move this models.__init__.py! I'll just add code comments too explaining what models they are currently used on.

I had kept it here because it's related to creating historical records in a specific pattern just used by notices and experiences, and experience code imports notice code in other ways too. But I also understand how by themselves they read fairly generic and don't want to cause confusion thinking they are solely for notices.